improver/utilities/mathematical_operations.py

```diff
@@ -356,3 +356,55 @@ def process(self, cube):
         integrated_cube = self.perform_integration(upper_bounds_cube, lower_bounds_cube)
 
         return integrated_cube
+
+
+def alinfit(x, y, axis=-1, gradient_only=False):
+    """Uses a simple linear fit approach to calculate the
+    gradient. Although equivalent, this
+    approach is much faster than using scipy lstsq to fit the
+    data.
+
+    Args:
+        x:
+            ndarray, x axis data
+        y:
+            ndarray, y axis data
+        axis:
+            Optional argument, specifies the axis to operate on.
+        gradient_only:
+            Boolean, if true only returns the gradient.
+
+    Returns:
+        The gradient and the y-intercept in 2 1d numpy arrays.
```

Comment:
Again, this doc-string needs improving please.

Reply:
Done